lib/shelley/test/unit/Cardano/Wallet/Shelley/CompatibilitySpec.hs

```diff
@@ -309,6 +316,159 @@ spec = do
+              , toSimpleScript $
+                  Cardano.RequireAllOf [toPaymentHash hashKeyTxt1, toPaymentHash hashKeyTxt2, toPaymentHash hashKeyTxt3]
+              )
+            , ( show version <> " RequireAnyOf"
+              , RequireAnyOf [toKeyHash hashKeyTxt1, toKeyHash hashKeyTxt2]
+              , toSimpleScript $
+                  Cardano.RequireAnyOf [toPaymentHash hashKeyTxt1, toPaymentHash hashKeyTxt2]
+              )
+            , ( show version <> " RequireAnyOf"
+              , RequireAnyOf [toKeyHash hashKeyTxt1, toKeyHash hashKeyTxt2, toKeyHash hashKeyTxt3]
+              , toSimpleScript $
+                  Cardano.RequireAnyOf [toPaymentHash hashKeyTxt1, toPaymentHash hashKeyTxt2, toPaymentHash hashKeyTxt3]
+              )
+            , ( show version <> " RequireSomeOf"
+              , RequireSomeOf 2 [toKeyHash hashKeyTxt1, toKeyHash hashKeyTxt2, toKeyHash hashKeyTxt3]
+              , toSimpleScript $
+                  Cardano.RequireMOf 2 [toPaymentHash hashKeyTxt1, toPaymentHash hashKeyTxt2, toPaymentHash hashKeyTxt3]
+              )
+            , ( show version <> " RequireSomeOf"
+              , RequireSomeOf 2 [toKeyHash hashKeyTxt1, toKeyHash hashKeyTxt2, toKeyHash hashKeyTxt3, toKeyHash hashKeyTxt4]
+              , toSimpleScript $
+                  Cardano.RequireMOf 2 [toPaymentHash hashKeyTxt1, toPaymentHash hashKeyTxt2, toPaymentHash hashKeyTxt3, toPaymentHash hashKeyTxt4]
+              )
+            , ( show version <> " nested 1"
+              , RequireSomeOf 2 [ toKeyHash hashKeyTxt1, toKeyHash hashKeyTxt2
+                                , RequireAllOf [toKeyHash hashKeyTxt3, toKeyHash hashKeyTxt4]
+                                ]
+              , toSimpleScript $
+                  Cardano.RequireMOf 2 [ toPaymentHash hashKeyTxt1, toPaymentHash hashKeyTxt2
+                                       , Cardano.RequireAllOf [toPaymentHash hashKeyTxt3, toPaymentHash hashKeyTxt4]
+                                       ]
+              )
+            , ( show version <> " nested 2"
+              , RequireAllOf [ toKeyHash hashKeyTxt1
+                             , RequireAnyOf [toKeyHash hashKeyTxt2, toKeyHash hashKeyTxt3, toKeyHash hashKeyTxt4]
+                             ]
+              , toSimpleScript $
+                  Cardano.RequireAllOf [ toPaymentHash hashKeyTxt1
+                                       , Cardano.RequireAnyOf [toPaymentHash hashKeyTxt2, toPaymentHash hashKeyTxt3, toPaymentHash hashKeyTxt4]
+                                       ]
+              )
+            , ( show version <> " nested 3"
+              , RequireSomeOf 1 [ toKeyHash hashKeyTxt1
+                                , RequireAllOf [ toKeyHash hashKeyTxt2
+                                               , RequireAnyOf [toKeyHash hashKeyTxt3, toKeyHash hashKeyTxt4 ]
+                                               ]
+                                ]
+              , toSimpleScript $
```

Comment:
These unit tests look good. Could the hspec title be changed to say that they are golden tests?